### PR TITLE
debug: fix compilation error for %p formatter

### PR DESCRIPTION
**Pull request by Yonezawa-T2**
Passing pointer value other than `void *` to `%p` formatter results in compilation error on Clang 7.3.0 on OS X:
```
RIOT/sys/net/network_layer/fib/fib.c:331:19: error: format specifies type 'void *' but
      the argument has type 'char *' [-Werror,-Wformat-pedantic]
                  msg.content.ptr, (int)i, (int)(table->notify_rp[i]));
                  ^~~~~~~~~~~~~~~
```

```
$ clang --version
Apple LLVM version 7.3.0 (clang-703.0.29)
Target: x86_64-apple-darwin15.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

This PR adds casts for pointers.

### CODE REVIEW

**Comment by OlegHahm** [comment]
Looks sane - @emmanuelsearch, @thomaseichinger, @kYc0o, can you verify?

**Comment by kYc0o** [comment]
I can :)

File: cpu/cc2538/periph/i2c.c [hunk 1]
```diff
@@ -188,7 +188,9 @@ static uint_fast8_t i2c_ctrl_blocking(uint_fast8_t flags)
 
     if (I2CM_STAT & BUSY) {
         /* If the controller is still busy, it probably will be forever */
+#ifdef MODULE_XTIMER
         DEBUG("Master is still BUSY after %u usec. Resetting.\n", xtimer_timeout);
+#endif
         cc2538_i2c_init_master(speed_hz);
```
Comment (kYc0o) [inline cpu/cc2538/periph/i2c.c]: is this part of this PR?

**Comment by kYc0o** [comment]
Can you rebase onto master? #5142 was merged.

**Comment by Yonezawa-T2** [comment]
Rebased.

File: cpu/cc2538/periph/i2c.c (hunk 1 above)
Comment (Yonezawa-T2) [inline cpu/cc2538/periph/i2c.c]: Strictly, not. Should I split this into another PR?

Comment (kYc0o) [inline cpu/cc2538/periph/i2c.c]: Yes, please.

Comment (Yonezawa-T2) [inline cpu/cc2538/periph/i2c.c]: Reverted the chunk.

**Comment by kYc0o** [comment]
ACK and go!

**Comment by kYc0o** [comment]
ups! I forgot the squash! (label was not set...) do I need to revert? @cgundogan @OlegHahm ?

**Comment by gebart** [comment]
there's no point in reverting and squashing, the unsquashed commits will still be visible in the history.

If you begin your commit message with the word "squash" then Murdock and Travis will be red even without the Github label set.

**Comment by OlegHahm** [comment]
No, don't revert. Shit happens. ;-)

**Comment by kYc0o** [comment]
Ok thanks! and sorry, I'll put more attention (more again...) in the future...

**Comment by OlegHahm** [comment]
No worries, also here you're not the first. :)

**Comment by authmillenon** [comment]
(deleted the revert branch in this repository)
